flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/TableSinkRestoreTest.java

```diff
@@ -44,6 +46,20 @@ public List<TableTestProgram> programs() {
                 TableSinkTestPrograms.SINK_WRITING_METADATA,
                 TableSinkTestPrograms.SINK_NDF_PRIMARY_KEY,
                 TableSinkTestPrograms.SINK_PARTIAL_INSERT,
-                TableSinkTestPrograms.SINK_UPSERT);
+                TableSinkTestPrograms.SINK_UPSERT,
+                TableSinkTestPrograms.INSERT_RETRACT_WITH_WRITABLE_METADATA_FOR_LEGACY_TYPE,
+                TableSinkTestPrograms.INSERT_RETRACT_WITH_WRITABLE_METADATA);
+    }
+
+    @Override
+    protected Stream<String> getSavepointPaths(
+            TableTestProgram program, ExecNodeMetadata metadata) {
+        // disable the writable metadata test for sink node with version 1. it fails after the
```

Comment:
Can we generalize this into a method in RestoreTestBase that TableSinkRestoreTest can implement for ignoring certain programs with versions?

Reply:
Not sure if that's what you had in mind, but I refactored it a bit.